apis/Google.Cloud.Storage.V1/Google.Cloud.Storage.V1.IntegrationTests/HmacKeysTest.cs

```diff
@@ -24,12 +24,19 @@ namespace Google.Cloud.Storage.V1.IntegrationTests
     [Collection(nameof(StorageFixture))]
     public class HmacKeysTest
     {
+        // https://cloud.google.com/storage/docs/authentication/hmackeys#restrictions
+        private static readonly TimeSpan s_hmacKeyCreationConsistencyDelay = TimeSpan.FromSeconds(30);
+        // https://cloud.google.com/storage/docs/consistency#strongly_consistent_operations
+        private static readonly TimeSpan s_hmacKeyStateChangeConsistencyDelay = TimeSpan.FromMinutes(3);
+
         private readonly StorageFixture _fixture;
 
         public HmacKeysTest(StorageFixture fixture) => _fixture = fixture;
 
         [Fact]
-        public void Lifecycle()
+        // This test is async because of the helpers for testing eventually consistent changes.
+        // We are testing sync production code though.
```

Comment:
Presumably we *could* just test the async code instead? Remove a Task.CompletedTask from each lambda...

Reply:
Alternatively, we could add an overload of EventuallyAsync that accepts an `Action` instead of a `Func<Task>`